Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -333,25 +448,49 @@ WarpXOpenPMDPlot::SaveRealProperty(WarpXParIter& pti,
 
 
 void
-WarpXOpenPMDPlot::SetupPos(openPMD::ParticleSpecies& currSpecies,
-               const unsigned long long& np) const
+WarpXOpenPMDPlot::SetupPos(const std::unique_ptr<WarpXParticleContainer>& pc,
+    openPMD::ParticleSpecies& currSpecies,
+    const unsigned long long& np) const
 {
-  auto particleLineup = openPMD::Dataset(openPMD::determineDatatype<amrex::ParticleReal>(), {np});
-
-  currSpecies["positionOffset"]["x"].resetDataset(particleLineup);
-  currSpecies["positionOffset"]["x"].makeConstant(0);
-  currSpecies["positionOffset"]["y"].resetDataset(particleLineup);
-  currSpecies["positionOffset"]["y"].makeConstant(0);
-  currSpecies["positionOffset"]["z"].resetDataset(particleLineup);
-  currSpecies["positionOffset"]["z"].makeConstant(0);
-
-  //auto positions = openPMD::Dataset(openPMD::determineDatatype<amrex::ParticleReal>(), {np});
-  currSpecies["position"]["x"].resetDataset(particleLineup);
-  currSpecies["position"]["y"].resetDataset(particleLineup);
-  currSpecies["position"]["z"].resetDataset(particleLineup);
-}
-
+  auto const realType = openPMD::Dataset(openPMD::determineDatatype<amrex::ParticleReal>(), {np});
+  auto const idType = openPMD::Dataset(openPMD::determineDatatype< uint64_t >(), {np});
+
+  for( auto const& comp : {"x", "y", "z"} ) {
+      currSpecies["positionOffset"][comp].resetDataset( realType );
+      currSpecies["positionOffset"][comp].makeConstant( 0. );
+      currSpecies["position"][comp].resetDataset( realType );
+      // meta data
+      //currSpecies["position"][comp].setUnitSI();
+      //currSpecies["positionOffset"][comp].setUnitSI();
```

Comment:
Are these lines still useful?

Reply:
the default, 1.0, works here, so I can remove the dead code. thx!